diff -Naur orig/cdc/build/share/defs.mk patched/cdc/build/share/defs.mk
--- orig/cdc/build/share/defs.mk	2012-03-24 16:54:56.000000000 +0100
+++ patched/cdc/build/share/defs.mk	2012-03-24 17:07:14.000000000 +0100
@@ -2353,7 +2353,7 @@
 SO_CC_CMD	= $(call compileCC,$(SO_CFLAGS),$@,$<)
 
 # SO_LINK_CMD(objFiles, extraLibs)
-SO_LINK_CMD 	= $(AT)$(TARGET_LD) $(SO_LINKFLAGS) -o $@ $(1) $(2)
+SO_LINK_CMD 	= $(AT)$(TARGET_LD) $(SO_LINKFLAGS) -o $@ $(1) $(LINKLIBS) $(2)
 JAVAC_CMD	= $(CVM_JAVAC) $(JAVAC_OPTIONS)
 JAR_CMD		= $(CVM_JAR)
 JAVA_CMD	= $(CVM_JAVA)
