diff -Naur orig/midp/src/events/input_port/fb/native/fb_read_key.c patched/midp/src/events/input_port/fb/native/fb_read_key.c
--- orig/midp/src/events/input_port/fb/native/fb_read_key.c	2011-03-09 18:57:24.000000000 +0100
+++ patched/midp/src/events/input_port/fb/native/fb_read_key.c	2011-04-01 21:14:15.105797350 +0200
@@ -77,10 +77,16 @@
 
 /** Update input keys state reading single char from keyboard device */
 jboolean read_char_key_event() {
-    unsigned char c;
+    unsigned short c;
     int readBytes = read(fbapp_get_keyboard_fd(), &c, sizeof(c));
     keyState.key = c;
     keyState.down = -1;
+
+    if (c == 0x0080) {
+        readBytes = read(fbapp_get_keyboard_fd(), &c, sizeof(c));
+        keyState.key = 0x010000 + c;
+    }
+
     return (readBytes > 0) ?
         KNI_TRUE : KNI_FALSE;
 }
