diff -Naur orig/midp/src/highlevelui/lcdlf/lfjava/classes/com/sun/midp/chameleon/skins/resources/SkinResourcesImpl.jpp patched/midp/src/highlevelui/lcdlf/lfjava/classes/com/sun/midp/chameleon/skins/resources/SkinResourcesImpl.jpp
--- orig/midp/src/highlevelui/lcdlf/lfjava/classes/com/sun/midp/chameleon/skins/resources/SkinResourcesImpl.jpp	2011-03-09 19:18:51.000000000 +0100
+++ patched/midp/src/highlevelui/lcdlf/lfjava/classes/com/sun/midp/chameleon/skins/resources/SkinResourcesImpl.jpp	2012-01-26 14:02:21.000000000 +0100
@@ -1,4 +1,12 @@
 /*
+ * NOTICE: Portions Copyright (c) 2007-2009 Davy Preuveneers.
+ * This file has last been modified by Davy Preuveneers on 2010-02-23. The
+ * changes are licensed under the terms of the GNU General Public
+ * License version 2. This notice was added to meet the conditions of
+ * Section 3.a of the GNU General Public License version 2.
+ */
+
+/*
  *   
  *
  * Copyright  1990-2007 Sun Microsystems, Inc. All Rights Reserved.
@@ -652,7 +660,7 @@
     /**
      * Cleanups after reading data from skin file
      */
-    private native static int finishReadingSkinFile()
+    private native static void finishReadingSkinFile()
         throws IOException;
 
     /**
